realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -111,7 +112,24 @@ public RealmObjectSchema setClassName(String className) {
         if (transaction.hasTable(internalTableName)) {
             throw new IllegalArgumentException("Class already exists: " + className);
         }
+        // in case if this table has a primary key, we need to transfer it after renaming the table.
+        String oldTableName = null;
+        String pkField = null;
+        if (table.hasPrimaryKey()) {
+            oldTableName = table.getName();
+            pkField = getPrimaryKey();
+        }
+        // this will throw when things go wrong
```

Comment:
Either make a more detailed description exactly how this could throw or remove it.